GeometricFiugures/src/main/java/Rectangle.java

```diff
@@ -0,0 +1,25 @@
+public class Rectangle extends Shape {
+
+    private float a;
+    private float b;
+
+    public Rectangle(float a, float b) {
+        this.a = a;
+        this.b = b;
+    }
+
+    public Rectangle(float a) {
```

Comment:
Jest ok, bo faktycznie prostokąt jest kwadratem. Można by pomysleć tylko, czy ze wzgledu na samo zadanie nie lepiej tutaj byłoby zostawić ten jeden konstruktor z fload a oraz float b, w po prostu w samym kwadracie użyc jego jako super ale przekazac mu po prostu dwa razy a:
    public Rectangle(float a, float b) {
        this.a = a;
        this.b = b;
    }
    
        public Square(float a) {
        super(a,a);
    }
    
wydaje mi się, że tutaj jest trochę inny przypadek niż z pracownikami kina. Ale ogólnie jest ok :)

Reply:
Faktycznie na pewno szybsze i zgrabniejsze :)

Reply:
w sumie to kwadrat jest prostokątem a nie prostokąt kwadratem